src/components/useMemo.jsx

```diff
@@ -1,25 +1,44 @@
-import { useState, useMemo, useEffect} from "react";
-
-export const Memo=()=>{
-
-      const [num,setNum]=useState(1);
-      
-      const costly=useMemo(()=>{
-        calc(num);
-      },[num])
-
-      function calc(n){
-        for(var i=0;i<10000000;i++){}
-        return (n*2);
-      }
-      //console.log(costly);
-    return (
-        <div>
-          <input placeholder="Enter Number" onChange={n=>{setNum(n.target.value)}}/>
-          <div>Input Number: {num}</div>
-          <div>Number after function execution: {costly} </div>
-          <p>First time loading will take some time, but second time loading will be faster than first load</p>
-        </div>
-    );
-    
-    }
\ No newline at end of file
+import { useState, useMemo } from "react";
+
+export const Memo = () => {
+  let [num, setNum] = useState(0);
+
+  let [col, setCol] = useState(false);
+
+  function costly(n) {
+    for (var i = 0; i < 1000000000; i++) {
+      n += 1;
+    }
+    return n;
+  }
+
+  const inc = () => {
+    setNum((n) => n + 1);
+  };
+
+  const colorChange = {
+    color: col === false ? "red" : "green",
+  };
+
+  //costly(num)
+
+  useMemo(() => {
```

Comment:
this is incorrect . For your program to be meaningful , we should store value returned from useMemo somewhere and this value shlould be a part of UI .

Reply:
Have added the suggested changes onto the useMemo branch